errors.go

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package rate
+
+import "errors"
+
+var (
+	// ErrCapacityReached indicates that the store cannot store any additional quotas.
+	ErrCapacityReached = errors.New("capacity reached")
+	// ErrLimitNotFound indicates that a limit could not be found for a given resource+action
+	ErrLimitNotFound = errors.New("limit not found")
+	// ErrInvalidParameter represents an invalid parameter error.
+	ErrInvalidParameter = errors.New("invalid parameter")
```

Comment:
You might want to consider having different errors for each invalid parameter. I think you would only need 3: `ErrInvalidLimit`, `ErrDuplicateLimit`, and `ErrInvalidNumBuckets`.

Reply:
I split out some separate errors. I needed more than just 3.